app/scripts/directives/realtimestring.js

```diff
@@ -353,25 +277,55 @@ angular.module('oncokbApp')
-                    // otherwise the function will be async which means return false directly.
-                    firebase.database().ref($scope.path).once('value', function(doc) {
-                        var data = doc.val();
-                        if (!_.isUndefined(data[$scope.key]) && !_.isEmpty(data[$scope.key])
-                            && data[$scope.key] === oldContent) {
-                            changedByOthers = true;
-                            return;
+                $scope.updateContent = function (n, o, isPropagation) {
+                    // 1) Do not run the function when no data change(n===o).
+                    // 2) Do not run the function when there is no new content(_.isUndefined(n)).
+                    // 3) Do not run the function when just click panel without any change(_.isEmpty(n) && _.isUndefined(o)).
+                    // 4) Do not run the function when file is not editable(scope.fe===false).
+                    if (n !== o && !_.isUndefined(n) && $scope.fe) {
+                        if (!$rootScope.reviewMode) {
+                            mainUtils.updateLastModified();
+                            if ($scope.t === 'treatment-select' && $scope.key === 'level' && !isPropagation) {
+                                $scope.changePropagation();
+                            }
+                            // 1) Do not trigger setReviewRelatedContent() when edit Additional Information (Optional).
+                            // 2) Do not trigger setReviewRelatedContent() when move mutations.
+                            if ($scope.key !== 'short') {
+                                $scope.setReviewRelatedContent(n, o, isPropagation);
+                            }
                         }
-                    }, function () {
-                        console.log('Failed to load firebase object');
-                    });
-                    return changedByOthers;
+                        if (n !== o && ($scope.key === 'level' || ['summary', 'diagnosticSummary', 'prognosticSummary'].includes($scope.key)  && $scope.mutation && $scope.tumor)) {
+                            $timeout(function() {
+                                $scope.indicateMutationContent($scope.mutation);
+                                $scope.indicateTumorContent($scope.tumor);
+                            }, 500);
+                        }
+                        if ($scope.t === 'p' || $scope.t === 'short') {
+                            $timeout.cancel($scope.timeoutRef);
+                            if ($scope.fe === true && !$scope.data[$scope.key+'_editing']) {
+                                $scope.data[$scope.key+'_editing'] = $rootScope.me.name;
+                            }
+                            $scope.timeoutRef = $timeout(function() {
+                                delete $scope.data[$scope.key+'_editing'];
+                                $scope.initializeFE();
+                            }, 10*1000);
```

Comment:
10s seems quite long time. What's the consideration here?

Reply:
I didn't change the time. It's set by Jiaojiao. I can reduce the time and test on it if you want.